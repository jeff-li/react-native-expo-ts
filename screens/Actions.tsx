import React from 'react';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { StyleSheet, Alert, Text } from 'react-native';

import Button from '../components/Button';
import { SafeViewBackground } from '../components/Background';

export default function Actions() {
  const mockData = [
    {
      id: "bd7acbea-c1b1-46c2-aed5-3ad53abb28ba",
      fullName: "Aafreen Khan",
      timeStamp: "12:47 PM",
      recentText: "Good Day!",
      avatarUrl:
        "https://images.pexels.com/photos/220453/pexels-photo-220453.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500",
    },
    {
      id: "3ac68afc-c605-48d3-a4f8-fbd91aa97f63",
      fullName: "Sujitha Mathur",
      timeStamp: "11:11 PM",
      recentText: "Cheer up, there!",
      avatarUrl:
        "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTyEaZqT3fHeNrPGcnjLLX1v_W4mvBlgpwxnA&usqp=CAU",
    },
    {
      id: "58694a0f-3da1-471f-bd96-145571e29d72",
      fullName: "Anci Barroco",
      timeStamp: "6:22 PM",
      recentText: "Good Day!",
      avatarUrl: "https://miro.medium.com/max/1400/0*0fClPmIScV5pTLoE.jpg",
    },
    {
      id: "68694a0f-3da1-431f-bd56-142371e29d72",
      fullName: "Aniket Kumar",
      timeStamp: "8:56 PM",
      recentText: "All the best",
      avatarUrl:
        "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSr01zI37DYuR8bMV5exWQBSw28C1v_71CAh8d7GP1mplcmTgQA6Q66Oo--QedAN1B4E1k&usqp=CAU",
    },
    {
      id: "28694a0f-3da1-471f-bd96-142456e29d72",
      fullName: "Kiara",
      timeStamp: "12:47 PM",
      recentText: "I will call today.",
      avatarUrl:
        "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRBwgu1A5zgPSvfE83nurkuzNEoXs9DMNr8Ww&usqp=CAU",
    },
  ]

  async function addMockData(): Promise<void> {
    try {
      await AsyncStorage.setItem('appData', JSON.stringify(mockData))
    } catch(e) {
      console.warn("Failed to add data.") 
    }
  }

  return (
    <SafeViewBackground>
      <Text style={styles.text}>Actions Screen</Text>
      <Button text='Add test data to Async Storage' onPress={addMockData} />
    </SafeViewBackground>
  );
}

const styles = StyleSheet.create({
  text: {
    color: '#333',
    fontSize: 24,
    fontWeight: 'bold'
  }
});