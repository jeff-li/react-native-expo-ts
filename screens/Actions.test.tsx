import React from "react";
import { render } from '@testing-library/react-native';

import Actions from "./Actions";

describe("<Actions />", () => {
  it("should match screenshot", () => {
    const { toJSON, getAllByText } = render(
      <Actions />
    );
    expect(toJSON()).toMatchSnapshot()
    const bananaElements = getAllByText('Actions Screen');
    expect(bananaElements).toHaveLength(1);
  });
});