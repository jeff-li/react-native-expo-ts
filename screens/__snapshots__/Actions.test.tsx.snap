// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Actions /> should match screenshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ebebeb",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    Actions Screen
  </Text>
</View>
`;
